app/query/[id]/page.tsx

```diff
@@ -0,0 +1,43 @@
+import QueryResult from "./QueryResult";
+import { Metadata } from "next";
+
+
+type MetaProps = {
+  params: { id: string }
+}
+
+
+export async function generateMetadata(
+    { params }: MetaProps
+): Promise<Metadata> {
+  // read route params
+  const id = params.id
+
+  // we want to get the data then can pass in as props
+
+  const res = await fetch(`https://www.asksalesforce.ai/api/query/${id}`, {
```

Comment:
You should be able to just call backend functions instead of running an API request here - should be significantly faster